### ISSUE TITLE
when "len(minibatch) == overflowed",  this line will return a empty list []. That cause a exception.

### ISSUE THREAD

**Issue by wang001** [OpenNMT/OpenNMT-py#1484]
https://github.com/OpenNMT/OpenNMT-py/blob/668c3ef362995c55633fde592354160fec1d1efd/onmt/inputters/inputter.py#L511

**Comment by wang001** [comment]
overflowed==1, batch_size==32

**Comment by guillaumekln** [comment]
Can you post a command line to reproduce this?

**Comment by wang001** [comment]
-data
E:/************
-save_model
E:/********
-layers
1
-rnn_size
1024
-word_vec_size
1024
-max_grad_norm
0
-optim
adam
-encoder_type
transformer
-decoder_type
transformer
-position_encoding
-dropout
0.2
-warmup_steps
8000
-learning_rate
2
-decay_method
noam
-label_smoothing
0.1
-adam_beta2
0.998
-batch_size
32
-batch_type
tokens
-normalization
tokens
-max_generator_batches
2
-train_steps
48500
-share_embeddings
-param_init_glorot
-world_size
1
-gpu_ranks
0
-valid_batch_size
16
-valid_steps
4000
-save_checkpoint_steps
4000
(I know this batch size is too small when batch_type is tokens.)

**Comment by vince62s** [comment]
This 32 token batch_size does not make sense.
However @francoishernandez we may test and make sure it does not yield an empty minibatch

**Comment by francoishernandez** [comment]
#1490 should fix it.
